src/Iterators/ResultIterator.php

```diff
@@ -35,25 +35,26 @@ class ResultIterator implements Iterator, Countable, SeekableIterator {
 	/**
 	 * @var boolean
 	 */
-	public $isResultWrapper = false;
+	public $numRows = false;
 
 	/**
 	 * @since 2.5
 	 *
-	 * @param ResultWrapper|array $res
+	 * @param Iterator|array $res
 	 */
 	public function __construct( $res ) {
 
-		if ( $res instanceof ResultWrapper ) {
-			$this->isResultWrapper = true;
+		if ( !$res instanceof Iterator && !is_array( $res ) ) {
```

Comment:
Probably more appropriate to check for `Traversable` instead of `Iterator`. Then you can also change the type hint in the doc to `iterable` (supertype of `Traversable` and `array`)